python/lsst/ci/hsc/gen2/validate.py

```diff
@@ -169,7 +169,19 @@ def checkPsfStarsAndFlags(self, catalog, minStellarFraction=0.95, doCheckFlags=T
         """Utility function for derived classes that want to verify PSF source
         selection and flag setting
         """
-        psfStarsUsed = catalog.get("calib_psf_used")
+        # pipe_task.propagateVisitFlags sets the flags PSF flag for
+        # all sources based on RA/DEC matching, so parent blends with
+        # children and failed blends are marked as PSF sources.
+        # There is also double counting when using scarlet, so we want
+        # to only check the undeblended isolated sources or
+        # the deblended models of isolated sources, but not both.
```

Comment:
I'm not 100% understanding the mapping of "undeblended isolated sources" and "deblended models of isolated sources" to the parent / nchild.  Can you expand on this a bit in the comments?